crates/c/src/lib.rs

```diff
@@ -1073,6 +1064,32 @@ void __wasm_export_{ns}_{snake}_dtor({ns}_{snake}_t* arg) {{
             self.src.h_defs(" ");
             self.print_typedef_target(id);
         }
+
+        // If target is a resource type, then we need to generate alias for own and borrow
+        // the own and borrow handle types must exist due to the generation step in `type_resource`.
+        if matches!(&self.resolve.types[target].kind, TypeDefKind::Resource) {
```

Comment:
We already have special handling logic in `type_alias`. See https://github.com/bytecodealliance/wit-bindgen/blob/main/crates/c/src/lib.rs#L1066-L1067

The reason that I added handles was mentioned above.

Reply:
This is related to an issue I used raised https://github.com/bytecodealliance/wit-bindgen/issues/763

Reply:
Ah yes the preexisting logic was what I was referring to as something I forgot, and ideally that logic would be removed as well since I don't think it should be necessary any more.

Reply:
I think this condition 
```rust
if !matches!(&self.resolve.types[target].kind,
                     TypeDefKind::Resource if self.gen.resources[&target].direction == Direction::Import) {
```
is still necessary because it prevents type_alias to generate `typedef` for `Resource` types as they are not used anywhere in the C gen. Only its `Handle `types are used.